### PR TITLE
Add Starship support

### PR DESCRIPTION
**Pull request by callummr**
🚀

### CODE REVIEW

**Comment by callummr** [comment]
The lint failures are on files I didn't touch..?

File: mackup/applications/starship.cfg [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[application]
+name = Starship
+
+[configuration_files]
```
Comment (lra) [inline mackup/applications/starship.cfg]: This should be moved under `xdg_configuration_files`, see https://github.com/lra/mackup/tree/master/doc#add-support-for-an-application-using-the-xdg-directory

Comment (callummr) [reply mackup/applications/starship.cfg]: Does this still apply when the application consuming the config file doesn't look for `XDG_CONFIG_HOME`, but hardcodes `.config` as the path it reads from? That's what Starship's doing here: https://github.com/starship/starship/blob/1eabd527252291398df5749d30f3459ea2a03823/src/config.rs#L184

Comment (lra) [reply mackup/applications/starship.cfg]: heh, no it doesn't.
Weird.

**Comment by hrvstr** [comment]
Am I doing something wrong here? I am using Mackup 0.8.28 (installed with home-brew) and I don't see starship when I do `mackup list`. I tried adding it to my config anyway and got this:

```
Traceback (most recent call last):
  File "/usr/local/bin/mackup", line 11, in <module>
    load_entry_point('mackup==0.8.28', 'console_scripts', 'mackup')()
  File "/usr/local/Cellar/mackup/0.8.28_1/libexec/lib/python3.8/site-packages/mackup/main.py", line 86, in main
    app = ApplicationProfile(mckp, app_db.get_files(app_name), dry_run, verbose)
  File "/usr/local/Cellar/mackup/0.8.28_1/libexec/lib/python3.8/site-packages/mackup/appsdb.py", line 141, in get_files
    return self.apps[name]["configuration_files"]
KeyError: 'starship'
```

**Comment by lra** [comment]
It's not in a released version yet, I plan to release one tonight.

**Pull request merged**